src/Runtime/OMExternalConstant.inc

```diff
@@ -158,7 +155,9 @@ bool omMMapBinaryFile(
     /* Make sure constAddr is the same as the mmap address.
     */
     if (constAddr[0] != tempAddr) {
```

Comment:
This check is wrong. Only the first thread that successfully performs the compare-and-swap will have `constAddr[0] == tempAddr`. All other threads will have `constAddr[0] != tempAddr`. But it doesn't matter since by design they will throw away their `tempAddr` and use the `constAddr[0]` set by the first thread.

Reply:
Does `tempAddr` become `NULL` after `munmap`? If so, we do the test only when `tempAddr != NULL`.

Reply:
I explicitly set `tempAddr = NULL` after `munmap`, and check `constAddr[0] != tempAddr` only when `tempAddr != NULL` (meaning for the first successful thread).

Reply:
Not sure what you mean. `munmap` cannot change `tempAddr` since C is passing by value (i.e., inside `munmap` it only has access to a copy of `tempAddr`). But even if `tempAddr` somehow becomes NULL after `munmap`, the check `constAddr[0] != tempAddr` is still wrong.

Reply:
> I explicitly set `tempAddr = NULL` after `munmap`, and check `constAddr[0] != tempAddr` only when `tempAddr != NULL` (meaning for the first successful thread).

Not sure why you want to do that. A successful compare-and-swap guarantees that `constAddr[0] == tempAddr`.

Reply:
Oh, you are right. In the latest code, I set `tempAddr = NULL` for failed threads, and check `if (tempAddr && (constAddr[0] != tempAddr))`. What do you think?

Reply:
I want to avoid the situation we encountered where `cds` was written wrongly. Perhaps, `assert` is better.

Reply:
OK. I removed the check. I was overthinking about having a check to debug, but since we found the issue, and `cds` would work correctly.

Reply:
Yes you can use `assert` but it wouldn't help in this case. Because even if you write your `cds` wrong like the bug we just had, in the first thread after the `cds` you will have `constAddr[0] == tempAddr`. Because `cds` doesn't know/care if the `tempAddr` you supplied in the call is garbage. All it does is setting `constAddr[0]` to `tempAddr` atomically, regardless of what's in `tempAddr`.